episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,743 @@
+](https://www.napari-hub.org/plugins/napari-simpleitk-image-processing). Both of 
+these plugins add many features that can aid with image segmentation and other 
+image processing tasks. Here, we've selected only these two to keep installation 
+fast and easy for this course. If you use Napari more in your own work though, 
+you may want to consider using [`devbio-napari`
+](https://www.napari-hub.org/plugins/devbio-napari) instead. `devbio-napari` 
+bundles many plugins into one useful package (including the two plugins we are 
+about to install!). There's detailed information on their [documentation pages
+](https://www.napari-hub.org/plugins/devbio-napari) if you're interested.
+
+In the top menu-bar of Napari select:  
+`Plugins > Install/Uninstall Plugins...`
+
+Then search for `napari-segment-blobs-and-things-with-membranes` and click the 
+blue button labelled 'install'. Wait for the installation to complete.
+
+![](fig/napari-segment-blobs-installation.png){alt="Screenshot of plugin 
+installation window for napari-segment-blobs-and-things-with-membranes"}
+
+Then search for `napari-simpleitk-image-processing` and click the blue install 
+button. Wait for the installation to complete.
+
+![](fig/napari-simple-itk-installation.png){alt="Screenshot of plugin 
+installation window for napari-simpleitk-image-processing"}
+
+Once both plugins are installed, **you will need to close and re-open Napari**.
+
+## Filters
+
+Let's open the cells image in Napari again:  
+`File > Open Sample > napari builtins > Cells (3D + 2Ch)`
+
+With the new plugins installed, you should see many new options under `Tools` in 
+the top menu-bar of Napari. You can find out more about these options using the 
+plugin's documentation which was linked in the [previous section
+](#adding-plugins-to-help-with-segmentation).
+
+For now, we are interested in the options under:  
+`Tools > Filtering / noise removal`
+
+Specifically, the two options labelled 'Gaussian' - you should see one ending in 
+'n-SimpleITK' and another ending 'scikit-image, nsbatwm'. These are two slightly 
+different implementations of a _gaussian blur_, one from 
+`napari-simpleitk-image-processing` and the other from 
+`napari-segment-blobs-and-things-with-membranes`. Both should work perfectly 
+well for our image, but let's use the one ending with 'scikit-image, nsbatwm' 
+for now.
+
+If you click on this option, you should see a new panel appear on the right side 
+of Napari:
+
+![](fig/gaussian-options.png){alt="Screenshot of settings for gaussian blur in 
+Napari" width='40%'}
+
+Make sure you have 'nuclei(data)' selected on the 'image' row, then click run:
+
+![](fig/nuclei-blurred-1.png){alt="Nuclei image after gaussian blur with sigma 
+of 1" width='40%'}
+
+You should see a new image appear in the layer list called 'Result of 
+gaussian_blur', which is a slightly blurred version of the original nuclei image.
+
+Try increasing the 'sigma' value to three and clicking run again:
```

Comment:
That's odd! For me, I can increase the sigma and click run multiple times. It always overwrites the previous option though - everything is written into the 'Result of gaussian_blur' layer, rather than creating new layers every time. Does yours not do that? If so, what kind of computer are you on - mac/windows/linux?